src/map_extras.cpp

```diff
@@ -76,6 +76,165 @@ void mx_null( map &, const tripoint & )
     debugmsg( "Tried to generate null map extra." );
 }
 
+void mx_roa_crash( map &m, const tripoint &abs_sub )
+{
+    int cx = rng( 6, SEEX * 2 - 7 );
+    int cy = rng( 6, SEEY * 2 - 7 );
+
+    for( int x = 0; x < SEEX * 2; x++ ) {
+        for( int y = 0; y < SEEY * 2; y++ ) {
+            if( m.veh_at( tripoint( x,  y, abs_sub.z ) ) &&
```

Comment:
This point is constructed over and over again here. I suggest either to iterate over `m.points_in_radius( tripoint( 0, 0, abs_sub.z ), 2 )` (this gives you `tripoint`s), or create the `tripoint` as local variable at the beginning of the second loop.

Reply:
I'll leave this one for later. You mean, replace the whole block of `if( m.veh_at( tripoint( x,  y, abs_sub.z ) ) &&` and subsequent code, with 'm.points_in_radius( tripoint( 0, 0, abs_sub.z ), 2 )`?

Reply:
At least create the tripoint once (as `const tripoint pnt( x, y, abs_sub.z );`) and use that tripoint in all places below instead of creating the tripoint again and again.

Or replace the two loops (over x and over y) with a single loop over the result of `point_in_radius`.